packages/atomic/src/components/commerce/atomic-commerce-pager/atomic-commerce-pager.spec.ts

```diff
@@ -0,0 +1,325 @@
+          store: {
+            state: {
+              resultList: {
+                focusOnFirstResultAfterNextSearch: () => {},
+              },
+            },
+          },
+          interfaceElement: {
+            type: 'product-listing',
+          },
+          i18n,
+        };
+        this.pagerState = {
+          page: 2,
+          totalPages: 10,
+          pageSize: 10,
+          totalEntries: 100,
+        };
+      }
+    };
+  }),
+  BindingController: class {},
+}));
+
+vi.mock('@coveo/headless/commerce', () => {
+  return {
+    buildProductListing: vi.fn(() => {
+      return {
+        pagination: vi.fn(() => ({
+          previousPage: vi.fn(),
+          selectPage: vi.fn(),
+          nextPage: vi.fn(),
+        })),
+      };
+    }),
+    buildSearch: vi.fn(() => {
+      return {
+        pagination: vi.fn(() => ({
+          previousPage: vi.fn(),
+          selectPage: vi.fn(),
+          nextPage: vi.fn(),
+        })),
+      };
+    }),
+  };
+});
+
+describe('AtomicCommercePager', () => {
+  let element: AtomicCommercePager;
+
+  const setupElement = async ({
+    numberOfPages,
+    mockedState = {
+      page: 0,
+      totalPages: 10,
+      pageSize: 10,
+      totalEntries: 100,
+    },
+  }: {numberOfPages?: number; mockedState?: PaginationState} = {}) => {
+    const element = await fixture<AtomicCommercePager>(
+      html`<atomic-commerce-pager></atomic-commerce-pager>`
+    );
+
+    if (numberOfPages !== undefined) {
+      element.setAttribute('number-of-pages', numberOfPages.toString());
+    }
+
+    element.pagerState = mockedState;
+    //@ts-expect-error - manually setting up this private property since a mocked implementation of createAppLoadedListener created problems. Couldn't manage keeping it to true when I needed it.
+    element.isAppLoaded = true;
+    element.initialize();
+
+    return element;
+  };
+
+  describe('when initializing', () => {
+    beforeEach(async () => {
+      element = await setupElement();
+    });
+
+    test('should validate props', () => {
+      const validatePropsSpy = vi.spyOn(element as never, 'validateProps');
+      element.initialize();
+      expect(validatePropsSpy).toHaveBeenCalled();
+    });
+
+    test('should call buildProductListing with engine when interfaceElement.type is product-listing', () => {
+      const buildProductListingMock = vi.spyOn(headless, 'buildProductListing');
+      element.bindings.interfaceElement.type = 'product-listing';
+
+      element.initialize();
+
+      expect(buildProductListingMock).toHaveBeenCalledWith(
+        element.bindings.engine
+      );
+    });
+
+    test('should call buildSearch with engine when interfaceElement.type is search', () => {
+      const buildSearchMock = vi.spyOn(headless, 'buildSearch');
+      element.bindings.interfaceElement.type = 'search';
+
+      element.initialize();
+
+      expect(buildSearchMock).toHaveBeenCalledWith(element.bindings.engine);
+    });
+  });
+
```

Comment:
I think you should also test that the `pager` is set to `searchOrListing.pager()` and that `createAppLoadedListener` is called.